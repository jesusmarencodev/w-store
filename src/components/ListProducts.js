import React from 'react';
import { useSelector } from 'react-redux';
import { Product } from './Product';

const ListProducts = () => {
  const { productList } = useSelector((state) => state.products);

  return (
    <article className="listProducts">
      <div className="shadow"></div>
      <div className="scrollbar">
        <div className="listProducts__list">
          {productList?.map((product) => (
            <Product key={product.id} product={product} />
          ))}
        </div>
      </div>
    </article>
  );
};

export default ListProducts;
