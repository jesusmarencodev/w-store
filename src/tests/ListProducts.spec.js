import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ListProducts from '../components/ListProducts';
import { products } from './mock/products';

const mockStore = configureStore([]);

describe('ListProducts component', () => {
  it('renders correctly', () => {
    const store = mockStore({
      products: {
        productList: products,
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <ListProducts />
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
