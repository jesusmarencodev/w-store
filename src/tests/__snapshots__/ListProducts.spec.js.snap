// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListProducts component renders correctly 1`] = `
<article
  className="listProducts"
>
  <div
    className="shadow"
  />
  <div
    className="scrollbar"
  >
    <div
      className="listProducts__list"
    >
      <div
        className="product d-flex a-center j-center   product--normal"
        onClick={[Function]}
      >
        <div
          className="product__quantity d-flex a-center j-center"
        >
          <span>
            1
          </span>
        </div>
        <img
          alt="Leche"
          className="product__img"
          src="alqueria.png"
        />
      </div>
      <div
        className="product d-flex a-center j-center   product--large"
        onClick={[Function]}
      >
        <div
          className="product__quantity d-flex a-center j-center"
        >
          <span>
            0
          </span>
        </div>
        <img
          alt="Detodito"
          className="product__img product__img--large d-flex"
          src="detodito.png"
        />
      </div>
      <div
        className="product d-flex a-center j-center   product--normal"
        onClick={[Function]}
      >
        <div
          className="product__quantity d-flex a-center j-center"
        >
          <span>
            0
          </span>
        </div>
        <img
          alt="Hit"
          className="product__img"
          src="1.png"
        />
      </div>
    </div>
  </div>
</article>
`;
